controllers/bucket_controller.go

```diff
@@ -188,3 +208,61 @@ func removeKey(slice []string, s string) []string {
 	}
 	return slice
 }
+
+func (b *BucketReconciler) WaitForSecret(namespace, name string) (*corev1.Secret, error) {
```

Comment:
Hate to be that guy, but can you use something from: https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait to do this instead?

Reply:
Done